OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseOutlineFragment.java

```diff
@@ -91,9 +105,101 @@ public void onItemClick(AdapterView<?> parent, View view, int position, long id)
             }
         });
 
+        // We only want long-press-to-delete functionality in only videos mode
+        if (isVideoMode) {
+            listView.setOnItemLongClickListener(new AdapterView.OnItemLongClickListener() {
+                @Override
+                public boolean onItemLongClick(AdapterView<?> parent, View view, int position, long id) {
+                    if (((IconImageView) view.findViewById(R.id.bulk_download)).getIcon() == FontAwesomeIcons.fa_check) {
+                        ((AppCompatActivity) getActivity()).startSupportActionMode(deleteModelCallback);
+                        listView.setItemChecked(position, true);
+                        adapter.getItem(position);
+                        return true;
+                    }
+                    return false;
+                }
+            });
+        }
+
         return view;
     }
 
+    /**
+     * Callback to handle the deletion of videos using the Contextual Action Bar.
+     */
+    private ActionMode.Callback deleteModelCallback = new ActionMode.Callback() {
+        // Called when the action mode is created; startActionMode() was called
+        @Override
+        public boolean onCreateActionMode(ActionMode mode, Menu menu) {
+            // Inflate a menu resource providing context menu items
+            MenuInflater inflater = mode.getMenuInflater();
+            inflater.inflate(R.menu.delete_contextual_menu, menu);
+            mode.setTitle(R.string.delete_videos_title);
+            return true;
+        }
+
+        // Called each time the action mode is shown. Always called after onCreateActionMode, but
+        // may be called multiple times if the mode is invalidated.
+        @Override
+        public boolean onPrepareActionMode(ActionMode mode, Menu menu) {
+            deleteMode = mode;
+            return false;
+        }
+
+        // Called when the user selects a contextual menu item
+        @Override
+        public boolean onActionItemClicked(ActionMode mode, MenuItem item) {
+            switch (item.getItemId()) {
+                case R.id.item_delete:
+                    final int checkedItemPosition = listView.getCheckedItemPosition();
+                    // Change the icon to download icon immediately
```

Comment:
Its not working, icon changed to download icon and then changed back to check icon instantly.